en/contributing/build.md

```diff
@@ -1,28 +1,71 @@
-# Building
+# Building DroneCore from Source
 
-## Build the Library
+To build all of DroneCore from source you will first need to build the C++ library, and then any wrappers. The instructions below show how.
 
-Make sure the submodules are updated first.
+## Build the C++ Library
 
+### Linux and Mac OS X
+
+First install the dependencies
+```bash
+sudo apt-get update -y
+sudo apt-get install cmake build-essential colordiff astyle git libcurl4-openssl-dev -y
 ```
+
+> **Note** If the build reports a missing dependency, confirm that the set above matches the requirements in the [Dockerfile](https://github.com/dronecore/DroneCore/blob/master/Dockerfile).
+
+Clone the DroneCore repository (or your fork) and update the submodules:
+
+```bash
+git clone https://github.com/dronecore/DroneCore.git
+cd DroneCore
 git submodule update --init --recursive
 ```
 
-### Desktop
-
-Build for the desktop to test and debug:
+You can then build the library by calling:
 
 ```
 make default
 ```
 
-### Android
+### Windows
+
+To build DroneCode on Windows you'll need to install Visual Studio Community 2015 with Update 3 (this is an older version, but is still available through the free Microsoft Dev Essentials Program).
+
+First download the [curl](https://curl.haxx.se/) source, extract and build it. These instructions assume you downloaded [curl-7.54.1.zip](https://curl.haxx.se/download/curl-7.54.1.zip) and extracted to the root of your C drive. The instructions can be modified for any other release.
+
+Open the *VS2015 x64 Native Tools Command Prompt*, go to the source directory and enter:
+
+```bash
+cd C:\curl-7.54.1\winbuild
+nmake /f Makefile.vc mode=static VC=15 MACHINE=x64 DEBUG=no
+```
+
+To build in Windows:
```

Comment:
@julianoes Is this "to build DroneCore"? And is this mkdir run in the the root of the DroneCore repo?